packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -12,19 +12,71 @@ import 'colors.dart';
 import 'constants.dart';
 import 'theme.dart';
 
-// Measured against iOS 12 in Xcode.
-const EdgeInsets _kButtonPadding = EdgeInsets.all(16.0);
-const EdgeInsets _kBackgroundButtonPadding = EdgeInsets.symmetric(
-  vertical: 14.0,
-  horizontal: 64.0,
-);
+
+// Measured against iOS 17 guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+
+/// The size of a [CupertinoButton].
+/// Based on the iOS Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+enum CupertinoButtonSize {
+  /// Displays a smaller button with round sides and smaller text (uses [CupertinoThemeData.textTheme.actionSmallTextStyle]).
+  small,
+  /// Displays a medium sized button with round sides and regular-sized text
+  medium,
+  /// Displays a (classic) large button with rounded edges and regular-sized text
+  large,
+}
+
+const Map<CupertinoButtonSize, EdgeInsetsGeometry> _kCupertinoButtonPadding = <CupertinoButtonSize, EdgeInsetsGeometry>{
```

Comment:
These maps that store raw style values should probably be made public. It makes it easier for developers making custom widgets to use, either by just referencing them in the docs or pulling them in directly in the code.

Reply:
We tend to make these kind of values private, and it's usually good practice to do so for writing app code, but we've received feedback that in the framework code it's a tad annoying.